akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -154,7 +148,46 @@ object Unmarshaller
         s"Right failure: ${right.getMessage}, " +
         s"Left failure: ${left.getMessage}")
 
+  /**
+   * Signals that unmarshalling failed because the entity content-type did not match one of the supported ranges.
+   * This error cannot be thrown by custom code, you need to use the `forContentTypes` modifier on a base
+   * [[akka.http.scaladsl.unmarshalling.Unmarshaller]] instead.
+   */
+  final case class UnsupportedContentTypeException(
+    supported:   Set[ContentTypeRange],
+    contentType: Option[ContentType]   = None)
+    extends RuntimeException(supported.mkString(
+      s"Unsupported Content-Type [$contentType], supported: ", ", ", "")) {
+
+    @deprecated("for binary compatibility")
+    def this(supported: Set[ContentTypeRange]) = this(supported, None)
+
+    @deprecated("for binary compatibility")
+    def copy(supported: Set[ContentTypeRange]): UnsupportedContentTypeException =
+      new UnsupportedContentTypeException(supported, this.contentType)
+
+    @deprecated("for binary compatibility")
+    def copy$default$1(supported: Set[ContentTypeRange]): UnsupportedContentTypeException =
+      new UnsupportedContentTypeException(supported, this.contentType)
+
+    @deprecated("for binary compatibility")
+    def copy(
+      supported:   Set[ContentTypeRange] = this.supported,
+      contentType: Option[ContentType]   = this.contentType): UnsupportedContentTypeException =
+      new UnsupportedContentTypeException(supported, contentType)
+  }
+
   object UnsupportedContentTypeException {
-    def apply(supported: ContentTypeRange*): UnsupportedContentTypeException = UnsupportedContentTypeException(Set(supported: _*))
+
+    @deprecated("for binary compatibility")
+    def apply(supported: ContentTypeRange*): UnsupportedContentTypeException =
+      new UnsupportedContentTypeException(Set(supported: _*), None)
+
+    @deprecated("for binary compatibility")
```

Comment:
yes, for compatibility but not really deprecated, the apply is perfectly fine as is.

Reply:
should i remove? as i was asked to add `deprecated` directly before